test/livebook/runtime/k8s_test.exs

```diff
@@ -0,0 +1,142 @@
+    System.cmd(
+      "kind",
+      ["export", "kubeconfig", "--name", @cluster_name, "--kubeconfig", @kubeconfig_path],
+      stderr_to_stdout: true
+    )
+
+    {_, bindings} = Code.eval_file("versions")
+
+    # build container image
+    System.cmd(
+      "docker",
+      ~w(buildx build --build-arg BASE_IMAGE=hexpm/elixir:#{bindings[:elixir]}-erlang-#{bindings[:otp]}-ubuntu-#{bindings[:ubuntu]} --build-arg VARIANT=default --load -t ghcr.io/livebook-dev/livebook:nightly .),
+      stderr_to_stdout: true
+    )
+
+    # load container image into Kind cluster
+    System.cmd(
+      "kind",
+      ["load", "docker-image", "--name", @cluster_name, "ghcr.io/livebook-dev/livebook:nightly"],
+      stderr_to_stdout: true
+    )
+
+    :ok
+  end
+
+  test "connecting flow" do
+    config = config()
+    req = req()
+
+    assert [] = list_pods(req)
+
+    pid = Runtime.K8s.new(config, req) |> Runtime.connect()
+
+    assert_receive {:runtime_connect_info, ^pid, "create pod"}, @assert_receive_timeout
+
+    assert_receive {:runtime_connect_info, ^pid, "waiting for pod"},
+                   @assert_receive_timeout
+
+    assert_receive {:runtime_connect_info, ^pid, "created container livebook-runtime"},
+                   @assert_receive_timeout
+
+    assert_receive {:runtime_connect_info, ^pid, "started container livebook-runtime"},
+                   @assert_receive_timeout
+
+    assert_receive {:runtime_connect_info, ^pid, "start proxy"}, @assert_receive_timeout
+    assert_receive {:runtime_connect_info, ^pid, "connect to node"}, @assert_receive_timeout
+    assert_receive {:runtime_connect_info, ^pid, "initialize node"}, @assert_receive_timeout
+    assert_receive {:runtime_connect_done, ^pid, {:ok, runtime}}, @assert_receive_timeout
+
+    Runtime.take_ownership(runtime)
+
+    assert [_] = list_pods(req)
+
+    # Verify that we can actually evaluate code on the Kubernetes Pod
+    Runtime.evaluate_code(runtime, :elixir, ~s/System.fetch_env!("TEST_VAR")/, {:c1, :e1}, [])
+    assert_receive {:runtime_evaluation_response, :e1, %{type: :terminal_text, text: text}, _meta}
+    assert text =~ "present"
+
+    Runtime.disconnect(runtime)
+
+    #  Wait for Pod to terminate (i.e status.phase == "Succeeded")
+    assert :ok ==
+             Kubereq.wait_until(
+               req,
+               "default",
+               runtime.pod_name,
+               &(&1["status"]["phase"] == "Succeeded")
+             )
+
+    # Finally, delete the Pod object.
+    Kubereq.delete(req, "default", runtime.pod_name)
+  end
+
+  defp req() do
+    [Kubereq.Kubeconfig.ENV, {Kubereq.Kubeconfig.File, path: @kubeconfig_path}]
+    |> Kubereq.Kubeconfig.load()
+    |> Kubereq.new("api/v1/namespaces/:namespace/pods/:name")
+  end
+
+  defp config(attrs \\ %{}) do
+    defaults = %{
+      context: "kind-#{@cluster_name}",
+      namespace: "default",
+      home_pvc: nil,
+      docker_tag: "nightly",
+      pod_template: %{template: @default_pod_template}
```

Comment:
```suggestion
      pod_template: @default_pod_template
```

The map is only used in the UI assigns, no?

Reply:
oh true - I forgot that the integration tests are excluded now.